browser/src/core/Socket.js

```diff
@@ -1187,6 +1187,19 @@ app.definitions.Socket = L.Class.extend({
 				this._map.zotero._onMessage(subTextMsg);
 			}
 		}
+		else if (textMsg.startsWith('updateroutetoken') && window.indirectionUrl != '') {
+			var http = new XMLHttpRequest();
```

Comment:
This looks fine - I wonder though if we can have a method on our Indirect websocket'y thing - that allows the refresh there ? so we can handle all that in one place (?) and again same comment on fetching the whole URL - in the future I'd hope we can switch from one server to a completely different one around the world with a different domain name =)